src/txn/txn_recover.c

```diff
@@ -823,6 +889,8 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
 done:
     WT_ERR(__recovery_set_checkpoint_timestamp(&r));
     WT_ERR(__recovery_set_oldest_timestamp(&r));
+    WT_ERR(__recovery_set_checkpoint_snapshot(&r));
```

Comment:
Can we move this function to the start of the recovery, as this retrieved snapshot is used in rollback to stable.

Reply:
Moved this function to the same place where we set the base write gen number.